modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java

```diff
@@ -142,8 +145,16 @@ public DDMFormEvaluatorEvaluateResponse evaluate() {
 
 		Stream<DDMFormRule> stream = ddmFormRules.stream();
 
+		Set<String> fieldNames = _getFieldNamesFromPage();
+
 		stream.filter(
 			DDMFormRule::isEnabled
+		).filter(
+			rule ->
+				(_ddmFormLayout == null) ||
+				(_ddmFormLayout.getPreviousPage() == null) ||
+				_conditionHasFieldName(rule.getCondition(), fieldNames) ||
```

Comment:
Hi @carolmariaabb , I've included the variable to avoid the method `_getFieldNamesFromPage` being called for each rule, but I can change it to remove `fieldNames`.